test/e2e/storage/utils/create.go

```diff
@@ -141,14 +141,7 @@ func PatchItems(f *framework.Framework, items ...interface{}) error {
 // - only the latest stable API version for each item is supported
 func CreateItems(f *framework.Framework, items ...interface{}) (func(), error) {
 	var destructors []func() error
-	var cleanupHandle framework.CleanupActionHandle
```

Comment:
I don't understand why this change was made - what's the difference here?

Reply:
The reason this change was made is because - cleanupActions are stored in a global map and this hook and hook installed in `framework.go` (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L186) might run out of order and what will happen is - if hook in framework.go ran before this hook then it will set `f.ClientSet` to nil - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L409 which will prevent this hook from working.

I am also thinking of making cleanupActions run in order (i.e last hook runs first)  but @thockin has tried in past to prevent hooks from being dependent on each other. 

There is one more reason why I have removed this code - This code here tears down CSI driver before testpod has been deleted. This obviously prevents testpod from cleaning up. We should rather let cleanup action delete the testpod and csidriver in order.

Reply:
FWIW - I have lost almost all context on how this works :)